ui/src/app/store/manage-collectiom/reducers.ts

```diff
@@ -160,10 +160,19 @@ const reducer: ActionReducer<CollectionEntityState> = createReducer(
   }),
 
   on(getSubjectExamplesFail, state => ({ ...state, isCollectionPending: false })),
-  on(addFileToCollection, (state, { url, file, subject }) => ({
-    ...state,
-    collection: [...state.collection, { url, file, subject, status: CircleLoadingProgressEnum.OnHold }],
-  })),
+
+  on(addFileToCollection, (state, { url, file, subject }) => {
+    const collectionCopy = [
+      ...state.collection.filter(item => item.status !== CircleLoadingProgressEnum.Failed && item.subject === state.subject),
```

Comment:
filter returns new array, no need to use [...state.collection.filter] construction, can be just state.collection.filter

Reply:
fixed